### DISCUSSION TITLE
Have Any Idea About Custom traits for Images

### DISCUSSION THREAD

**Discussion by AgnRakul** [GrapesJS/grapesjs#4737] [General]
![image](https://user-images.githubusercontent.com/70213104/202629025-800ecc45-862e-4cdb-9670-99b4c6f625be.png)
Custom Traits for Images Like the Above Image

**Comment by artf** [comment]
You can create your [custom traits](https://grapesjs.com/docs/modules/Traits.html#define-new-trait-type).

**Reply by AgnRakul** [reply]
I have Created The Traits but While Changing the Src its not applying on image tag its just changing in traits preview

**Comment by zeeshan-algo** [comment]
@AgnRakul  Did you solve this problem?

**Reply by AgnRakul** [reply]
Yep

**Reply by zeeshan-algo** [reply]
can u plzz tell how ?

**Reply by zeeshan-algo** [reply]
this is my code 
{
      name: 'image',
      type: 'file',
      label: 'Icon',
      attributes: {
        accept: 'image/*',
        onchange: function () {
          let url = this.value;
          console.log('image value------------------------>', url);
          let img = new Image();
          img.src = url;
          img.onload = function () {
            editor.AssetManager.add({ src: url });
            editor.getSelected().set('image', url);
            // Trigger a render of the selected component to update the canvas
            //@ts-ignore
            editor.getSelected().trigger('change:attributes');
          };
        },
      },
    },
